### PR TITLE
feat(dlp): added a sample for deidentify redact

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [deidentify redact](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-redact) with unit test case.

### CODE REVIEW

File: dlp/snippets/deid/deid_test.go [hunk 1]
```diff
@@ -100,3 +100,24 @@ func TestDeidentifyDateShift(t *testing.T) {
...
 }
+
+func TestDeIdentifyWithRedact(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	input := "My name is Alicia Abernathy, and my email address is aabernathy@example.com."
+	infoTypeNames := []string{"EMAIL_ADDRESS"}
+	want := "output: My name is Alicia Abernathy, and my email address is ."
+
+	t.Run(input, func(t *testing.T) {
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:111]: question: why are `t.Run()` and `t.Parallel()` being used here? It looks like there's only one test case. Please remove if not needed.

File: dlp/snippets/deid/deid_with_redact.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:46]: issue: remove `var`, use `:=`.

File: dlp/snippets/deid/deid_with_redact.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		DataItem: &dlppb.ContentItem_Value{
+			Value: inputStr,
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:52]: praise: excellent use of comment to direct back to docs.

File: dlp/snippets/deid/deid_with_redact.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+	var infoTypes []*dlppb.InfoType
+	for _, it := range infoTypeNames {
+		infoTypes = append(infoTypes, &dlppb.InfoType{Name: it})
+	}
+	var inspectConfig = &dlppb.InspectConfig{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:58]: issue: remove `var`, use `:=`.

File: dlp/snippets/deid/deid_with_redact.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+		InfoTypes: infoTypes,
+	}
+
+	// Define type of de-identification.
+	var primitiveTransformation = &dlppb.PrimitiveTransformation{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:63]: issue: remove `var`, use `:=`.

File: dlp/snippets/deid/deid_with_redact.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+		Transformation: &dlppb.PrimitiveTransformation_RedactConfig{
+			RedactConfig: &dlppb.RedactConfig{},
+		},
+	}
+
+	// Associate de-identification type with info type.
+	var transformation = &dlppb.InfoTypeTransformations_InfoTypeTransformation{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:70]: issue: remove `var`, use `:=`.

File: dlp/snippets/deid/deid_with_redact.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		InfoTypes:               infoTypes,
+		PrimitiveTransformation: primitiveTransformation,
+	}
+
+	// Construct the configuration for the Redact request and list all desired transformations.
+	var redactConfig = &dlppb.DeidentifyConfig{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:76]: issue: remove `var`, use `:=`.

File: dlp/snippets/deid/deid_with_redact.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+		Parent:           fmt.Sprintf("projects/%s/locations/global", projectID),
+		DeidentifyConfig: redactConfig,
+		InspectConfig:    inspectConfig,
+		Item:             contentItem,
+	}
+
+	// Send the request.
+	resp, err := client.DeidentifyContent(ctx, req)
+	if err != nil {
+		return fmt.Errorf("DeidentifyContent: %v", err)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_redact.go:97]: issue: don't wrap the error, just return it as-is.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid/deid_test.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Changed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go:46]: fixed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: fixed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: fixed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 7 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: fixed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 8 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: Point noted,  fixed it.

File: dlp/snippets/deid/deid_with_redact.go (hunk 2 above)
Comment (telpirion) [reply dlp/snippets/deid/deid_with_redact.go:46]: It doesn't look like this was fixed.

File: dlp/snippets/deid/deid_with_redact.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go:52]: Thank you 🙂

File: dlp/snippets/deid/deid_with_redact.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: Yup, my bad.
now it's fixed.

Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_redact.go]: yup, that's my bad, So sorry for that. Now it's fixed.

**Review by telpirion — APPROVED** [review]

**Pull request merged**